### ISSUE TITLE
inet data type error with IPv6 input string

### ISSUE THREAD

**Issue by tzach** [scylladb/scylla#3187]
*Installation details*
Scylla version (or git commit hash): 2.0

The following works with Apache Cassandra 2.2:
```
CREATE KEYSPACE mykeyspace WITH REPLICATION = { 'class' : 'SimpleStrategy', 'replication_factor' : 1 };
use mykeyspace;
CREATE TABLE myips (
    mac_addrs text PRIMARY KEY,
    server_ip inet
);
INSERT INTO myips (mac_addrs, server_ip ) VALUES ('abcede', '10.0.2.5');
INSERT INTO myips (mac_addrs, server_ip ) VALUES ('abcede', '2002:4559:1FE2::4559:1FE2');
```

On Scylla 2.0, last row return
```
cqlsh:mykeyspace> INSERT INTO myips (mac_addrs, server_ip ) VALUES ('abcede',
              ... '2002:4559:1FE2::4559:1FE2');
InvalidRequest: Error from server: code=2200 [Invalid query] message="marshalling error"
```

**Comment by tzach** [comment]
@penberg ?

**Comment by eyalgutkind** [comment]
Any update on this issue?
/cc @gmatza

**Comment by elcallio** [comment]
While it is fairly easy to replace the use if ipv4_addr as native type (and consequent non-support for ipv6) in the actual data type used in scylla, it would break all the code (various places) that put and retrieve inet cells in tables, since these places right now assume and use ipv4-only address types. And since the connection cql->what comes out in our native code is somewhat loose, it would be tricky to make sure all places are dealt with. 
Might need to change gms::inet_address to use seastar::net::inet_address, and ensure interop with all networking to do this safely.

**Comment by tzach** [comment]
@elcallio note that this is for CQL only, not for seastar/scylla itself to support ipv6 (#2027)
Does this make it easier?

**Comment by elcallio** [comment]
No. Scylla still writes/reads ipv4 addresses from/to tables. But it might be possible to augument reasonably semi-intrusively. I just worry about places that might do the equivalent of reinterpret_cast.

**Comment by ysubach** [comment]
I'm seeing same error `code=2200 [Invalid query] message="marshalling error"`.
Is this fix available in version 2.2.0?

**Comment by tzach** [comment]
@ysubach look like the commit was merged to 2.3, not 2.2
https://github.com/scylladb/scylla/commit/b1edf75c8bc6334fa6481202f15b43d4dc6063ff

**Comment by ysubach** [comment]
Ok, thanks for letting me know.
